lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -3045,21 +3052,33 @@ void FEnumCreateOp::print(OpAsmPrinter &printer) {
   printer << '(' << getInput() << ')';
   SmallVector<StringRef> elidedAttrs = {"fieldIndex"};
   printer.printOptionalAttrDictWithKeyword((*this)->getAttrs(), elidedAttrs);
-  printer << " : " << getResult().getType();
+  printer << " : ";
+  printer.printFunctionalType(ArrayRef<Type>{getInput().getType()},
+                              ArrayRef<Type>{getResult().getType()});
 }
 
 ParseResult FEnumCreateOp::parse(OpAsmParser &parser, OperationState &result) {
   auto *context = parser.getContext();
 
   OpAsmParser::UnresolvedOperand input;
   std::string fieldName;
-  Type outputType;
+  mlir::FunctionType functionType;
```

Comment:
I changed `enumcreate` to use a functional type to allow for const operands for a non-const result. It also matches the other create ops.